App.js

```diff
@@ -26,29 +20,39 @@ const storeData = async (value) => {
 
 const getData = async () => {
 	  try {
-	const jsonValue = await AsyncStorage.getItem('@tasks')
-	 return jsonValue != null ? JSON.parse(jsonValue) : null;
-        } catch(err) {
-		console.log(err);
+      const jsonValue = await AsyncStorage.getItem('@tasks')
+      
+      console.log('inside get promise', jsonValue, JSON.parse(jsonValue))
+      return jsonValue != null ? JSON.parse(jsonValue) : null;
+    } catch(err) {
+      console.log(err);
 	  }
-	}
-			    
+}
+  
+ const addTodo = () => {
+  let id = todos[todos.length - 1]?.id || 0 ;
+   let thisId = id * 1 + 1 + "";
+   const updatedToDoArray = [...todos, { id: thisId, text }];
+  setTodos(updatedToDoArray);
+  storeData(updatedToDoArray)
```

Comment:
I moved `storeData` to the function which adds a todo. That way every time someone adds a todo the array of todos stored will be updated. You had your storeData in a `useEffect` hook, which had an empty dependency array, which means the function run ONLY when the component was mounted. I invite you to read more about the `useEffect` hook in the react docs: https://reactjs.org/docs/hooks-reference.html#useeffect